soroban-sdk/src/tests/storage_testutils.rs

```diff
@@ -0,0 +1,42 @@
+use crate::{
+    self as soroban_sdk,
+    testutils::storage::{Instance as _, Persistent as _, Temporary as _},
+    Map, Val,
+};
+use soroban_sdk::{contract, Env};
+
+#[contract]
+pub struct Contract;
+
+#[test]
+fn all() {
+    let e = Env::default();
+    let id = e.register_contract(None, Contract);
+
+    e.as_contract(&id, || {
+        e.storage().instance().set(&1, &2);
+        e.storage().instance().set(&1, &true);
+        e.storage().instance().set(&2, &3);
+        e.storage().persistent().set(&10, &20);
+        e.storage().persistent().set(&10, &false);
+        e.storage().persistent().set(&20, &30);
+        e.storage().temporary().set(&100, &200);
+        e.storage().temporary().set(&100, &());
+        e.storage().temporary().set(&200, &300);
+    });
+
+    e.as_contract(&id, || {
+        assert_eq!(
+            e.storage().instance().all(),
+            Map::<Val, Val>::from_array(&e, [(1.into(), true.into()), (2.into(), 3.into())])
```

Comment:
Note that these `.all()` functions, like any storage function, must be called in the context of a contract.